modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/main/resources/META-INF/resources/admin/view_form_instance.jsp

```diff
@@ -107,34 +107,6 @@ FormInstancePermissionCheckerHelper formInstancePermissionCheckerHelper = ddmFor
 											cssClass="table-cell-expand table-title"
 											name="name"
 										>
-
-											<%
-											String errorMessage = StringPool.BLANK;
-
-											if (!hasValidDDMFormFields) {
-												errorMessage = LanguageUtil.format(request, "this-form-was-created-using-a-custom-field-type-x-that-is-not-available-for-this-liferay-dxp-installation.-instal-x-to-make-it-available-for-editing", ddmFormAdminDisplayContext.getInvalidDDMFormFieldType(formInstance));
-											}
-											else if (!hasValidMappedObject) {
-												if (objectDefinition == null) {
-													errorMessage = "this-form-was-created-using-a-deleted-object-as-storage-type";
-												}
-												else {
-													errorMessage = LanguageUtil.format(request, "this-form-was-created-using-an-inactive-object-as-storage-type.-activate-x-object-to-make-it-available-for-editing", ddmFormAdminDisplayContext.getObjectDefinitionLabel(formInstance, locale));
-												}
-											}
-											else if (!hasValidStorageType) {
-												errorMessage = LanguageUtil.format(request, "this-form-was-created-using-a-storage-type-x-that-is-not-available-for-this-liferay-dxp-installation.-install-x-to-make-it-available-for-editing", formInstance.getStorageType());
-											}
-											%>
-
-											<span class="error-icon">
-												<liferay-ui:icon
-													icon="exclamation-full"
-													markupView="lexicon"
-													message="<%= errorMessage %>"
-													toolTip="<%= true %>"
-												/>
-											</span>
```

Comment:
@fmarcio thanks for your work on this.. But unfortunately we can't delete this code, as it will change the expected behavior for Forms. 

After reading the ticket description I noticed that the error must happen because after deleting and importing the object definition, the `ID` will change, but the imported form is associated with the `ID` of the deleted object.

If you read the comments in the ticket, you will see the following proposed solution for this bug:

> In a quick conversation with [Gabriel Albuquerque](https://issues.liferay.com/secure/ViewProfile.jspa?name=gabriel.albuquerque), he raised that it must be corrected on the Forms side and it may be the case of referencing the External Reference Code instead of the Object Id. Correct me if I'm wrong, Gabriel.

The correct fix is to associate the Form with the `externalReferenceCode` of the Object, and not the `ID`.

That said, I'll close this PR. 

Please let me know if there's anything I can do to help.